Dockerfile

```diff
@@ -39,15 +35,11 @@ RUN export TAG=$(curl --silent "https://api.github.com/repos/kubernetes-sigs/kus
     curl -SL "https://github.com/kubernetes-sigs/kustomize/releases/download/${TAG}/kustomize_${VERSION_TAG}_${OS}_${ARCH}.tar.gz" > kustomize.tgz && \
     tar -xvf kustomize.tgz
 
-FROM golang:1.21.3 as build
-WORKDIR /app
-COPY . .
-RUN make static
```

Comment:
Be nicer if this could be kept, so a `docker build` continues to work as expected and isn't a two-step thing where if you forget to run `make static` you can get an outdated binary in the image.

It should be possible to do this with `make`, since it can avoid rebuilding the target if the inputs haven't changed. Would require a change in `Makefile` to list the dependencies (`*.go`, `go.mod` and `go.sum`) but shouldn't be too bad to do.